data_format.md

```diff
@@ -208,3 +208,96 @@ centroids = dt.agents[slice(10_000)]["centroid"]  # note this is the same as dt.
 we reduce the decompression numbers by **a factor of 10K**.
 
 **TL;DR**: when working with `zarr` you should always aim to minimise the number of accesses to the compressed data.
+
+## Dataset Abstraction Classes
+
+As shown above, working with the raw `zarr` dataset has its own perils. To that end, we provide two structures
+which form an additional abstraction layer over the raw `zarr` dataset. These two python classes allow to rasterise
+and get information about the past and future state of the AV or another agent. 
+
+**Note:** the following 2 classes inherit from Pytorch Dataset and as such are tied to work with it.
+
+**Note:** the following 2 classes assume the world to be rasterised as BEV (Bird-Eye-View), this is a common choice for 
+CNN-based approaches. Still, this can be disabled by using `stub_debug` as `map_type`.
+
+
+### EgoDataset
+The `EgoDataset` retrieves information about the status of the AV in the current frame and the frames before it (if history is enabled).
+It can be iterated over the frame annotations to return a dict with the following fields:
```

Comment:
This sentence doesn't make sense: 'It can be iterated over the frame annotations to return a dict with the following fields' - what do you mean?

Reply:
that if you iterate over it you get a dict with those fields. I will rephrase it :)